pkg/kubelet/container/types.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"errors"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+var (
+	ErrNoSuchImage = errors.New("no such image")
+)
+
+// Container represents a container.
+type Container struct {
+	// The id of the container.
+	ID string `json:"id"`
+	// The name of the container.
+	Name string `json:"name,omitempty"`
+	// The host name of the container.
+	Hostname string `json:"hostname,omitempty"`
+	// The name of the container's image.
+	Image string `json:"image,omitempty"`
+	// The id of the container's image.
+	ImageID string `json:"image,omitempty"`
+	// The State of the container, including when
+	// the container is created/started.
+	State State `json:"state,omitempty"`
+	// The network settings of the container, including
+	// the ip of the container and its port bindings.
+	NetworkSettings *NetworkSettings `json:"networkSettings`
+	// The volumes binded to the container.
+	Volumes map[string]string `json:"volumes,omitempty"`
+	// The environment variables of the container.
+	Env []string `json:"env,omitempty"`
+}
+
+// State represents the state of a container.
```

Comment:
Except for removal of the yaml tags, this is a copy+paste of https://github.com/fsouza/go-dockerclient/blob/master/container.go#L89. 

I know I asked you to address the issue of Dockerisms in this API, but this PR doesn't do that. For now, probably it would be better to go back to using dockerclient types directly, until we figure out what API we really want, as per @thockin's questions (deleting unused fields, etc.). 

Future use cases we already know we'll likely want to support:
* Pure Java applications (Fabric8)
* Non-Linux platforms (MacOS, BSD, Windows)
* Legacy (pre-Docker) container runtimes and package formats